tests/test_apply_variants.py

```diff
@@ -498,111 +631,188 @@ def test_insertion_at_interval_ends(self):
-        interval = Interval('chr1', '+', 5, 15, genome37, 0)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 0)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 0)
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, (5, 0), (5, 1), 5, 6, 7])
 
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [2, 3, 4, (5, 0), (5, 1), 5, 6, 7, 8, 9])
+
         # Test anchor right of deletion
         variants = [Variant.from_string("chr1:11:GT:", self.genome)]
-        interval = Interval('chr1', '+', 5, 15, genome37, 11)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
+
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 11)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 11)
+
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
         self.assertEqual(reference_alignment, [-1, 0, 1, 2, 3, 4, 6, 7, 8, 9])
 
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [0, 1, 2, 3, 5, 6, 7, 8, 9, 10])
+
         # Test anchor right of insertion
         variants = [Variant.from_string("chr1:11::TT", self.genome)]
-        interval = Interval('chr1', '+', 5, 15, genome37, 15)
-        reference_alignment = apply_variants(genome37.dna, variants, interval, reference_alignment=True)[1]
-        self.assertEqual(reference_alignment, [2, 3, 4, (5, 0), (5, 1), 5, 6, 7, 8, 9])
 
-    def test_reference_alignment_exception(self):
-        genome37 = MiniGenome('test_genome')
-        variants = [Variant.from_string("chr1:11:GTA:ATG", genome37)]
-        interval = Interval('chr1', '-', 5, 15, genome37, 5)
+        positive_strand_interval = Interval('chr1', '+', 5, 15, genome37, 15)
+        negative_strand_interval = Interval('chr1', '-', 5, 15, genome37, 15)
 
-        with self.assertRaises(ValueError):
-            apply_variants(genome37.dna, variants, interval, reference_alignment=True)
+        reference_alignment = apply_variants(genome37.dna, variants, positive_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [2, 3, 4, (5, 0), (5, 1), 5, 6, 7, 8, 9])
+
+        reference_alignment = apply_variants(genome37.dna, variants, negative_strand_interval, reference_alignment=True)[1]
+        self.assertEqual(reference_alignment, [0, 1, 2, 3, 4, (5, 0), (5, 1), 5, 6, 7])
```

Comment:
this doesn't seem to be correct,

if you deleted before the anchor on the positive strand, you'd be deleting after the anchor on the negative strand.

in both cases you would be sliding new characters from the start of the chromosome